rust/src/smb/files.rs

```diff
@@ -89,6 +89,34 @@ impl SMBState {
         return tx_ref.unwrap();
     }
 
+    /// get file tx for a open file. Returns None if a file for the fuid exists,
+    /// but has already been closed.
+    pub fn get_file_tx_by_fuid_with_open_file(&mut self, fuid: &[u8], direction: Direction)
+        -> Option<&mut SMBTransaction>
+    {
+        let f = fuid.to_vec();
+        for tx in &mut self.transactions {
```

Comment:
Do we really need this loop over transactions ?

Reply:
Goal is to look up a tx in the tx vector, so I think so.

Reply:
> Goal is to look up a tx in the tx vector, so I think so.

Wrong account, wh00ps